specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json

```diff
@@ -0,0 +1,332 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2024-05-01-preview",
+    "title": "SqlManagementClient",
+    "description": "The Azure SQL Database management API provides a RESTful set of web APIs that interact with Azure SQL Database services to manage your databases. The API enables users to create, retrieve, update, and delete databases, servers, and other entities."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/operations": {
+      "get": {
+        "tags": [
+          "DatabaseOperations"
+        ],
+        "description": "Gets a list of operations performed on the database.",
+        "operationId": "DatabaseOperations_ListByDatabase",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "description": "The name of the resource group that contains the resource.",
+            "required": true,
+            "type": "string",
+            "maxLength": 90,
+            "minLength": 1,
+            "pattern": "^[\\p{L}\\p{N}_().-]*[\\p{L}\\p{N}_()-]$"
+          },
+          {
+            "name": "serverName",
+            "in": "path",
+            "description": "The name of the server.",
+            "required": true,
+            "type": "string",
+            "maxLength": 63,
+            "minLength": 1,
+            "pattern": "^[a-z0-9]+(-*[a-z0-9]+)*$"
+          },
+          {
+            "name": "databaseName",
+            "in": "path",
+            "description": "The name of the database.",
+            "required": true,
+            "type": "string",
+            "maxLength": 128,
+            "minLength": 1,
+            "pattern": "^[^<>*%&:\\\\\\/?\\x00-\\x1F]*[^.<>*%&:\\\\\\/?\\s\\x00-\\x1F]$"
```

Comment:
>"pattern": "^[^<>*%&:\\\\\\/?\\x00-\\x1F]*[^.<>*%&:\\\\\\/?\\s\\x00-\\x1F]$" [](http://example.com/codeflow?start=12&length=76)

if you are adding patterns now, will the users who have databaseName outside this pattern, cannot use the new apiversion then ?

Reply:
I assume the pattern is chosen for backwards compatibility